genenetwork/genenetwork2
# Feature/add UI for editing metadata

#### Description

This PR:
- Remove logger info from [webqtlDatabaseFunction](https://github.com/genenetwork/genenetwork2/pull/571/commits/002108e14f3d3823c43d7faebaf6f1f8b3c7a3df)
- Replaces `get_species` with `get_all_species` from gn3
- Adds edit trait capability(you can't edit individual traits now)
- Adds an login decorator
- Updates the docker image to point to: [genenetwork2](https://hub.docker.com/repository/docker/bonfacekilz/genenetwork2)

#### How should this be tested?

- All tests should pass
- @robwwilliams on any trait page, you should be able to see the `Edit` button. Clicking that takes you to the edit page(only @robwwilliams has access to this page).

#### Any background context you want to provide?

Most of that work is done. Improvements that'll be added later:
- Storing a diff of edits.
- Thinking out redirects for pages not accessible to users: Now I simply just show a message: `return "You need to be admin", 401`

#### What are the relevant pivotal tracker stories?

N/A

#### Screenshots (if appropriate)
Edit button:
![image](https://user-images.githubusercontent.com/11820306/119638312-32c58080-be1f-11eb-96fa-114ee4937d67.png)

Edit page:
![image](https://user-images.githubusercontent.com/11820306/119639006-ddd63a00-be1f-11eb-9ee6-1c7b7afde5dc.png)


#### Questions

N/A
